java/splitter/splitter-core/src/main/java/sleeper/splitter/FindPartitionsToSplit.java

```diff
@@ -43,28 +43,26 @@
  */
 public class FindPartitionsToSplit {
     private static final Logger LOGGER = LoggerFactory.getLogger(FindPartitionsToSplit.class);
-    private final TablePropertiesProvider tablePropertiesProvider;
     private final StateStoreProvider stateStoreProvider;
     private final JobSender jobSender;
     private final int maxFilesInJob;
 
     public FindPartitionsToSplit(
             InstanceProperties instanceProperties,
-            TablePropertiesProvider tablePropertiesProvider,
             StateStoreProvider stateStoreProvider,
             JobSender jobSender) {
-        this.tablePropertiesProvider = tablePropertiesProvider;
         this.stateStoreProvider = stateStoreProvider;
         this.jobSender = jobSender;
         this.maxFilesInJob = instanceProperties.getInt(MAX_NUMBER_FILES_IN_PARTITION_SPLITTING_JOB);
     }
 
-    public void run() {
-        tablePropertiesProvider.streamOnlineTables().forEach(tableProperties -> {
+    public void run(Stream<TableProperties> tables) {
+        tables.forEach(table -> {
             try {
-                findPartitionsToSplit(tableProperties, stateStoreProvider.getStateStore(tableProperties));
+                findPartitionsToSplit(table, stateStoreProvider.getStateStore(table));
             } catch (StateStoreException e) {
-                LOGGER.error("StateStoreException thrown whilst running FindPartitionsToSplit", e);
+                LOGGER.error("StateStoreException thrown whilst running FindPartitionsToSplit for table {}",
+                        table.getStatus(), e);
```

Comment:
Ideally the lambda should fail if any exception is thrown here. It should also still move onto the next table regardless of what the failure was, not just for a StateStoreException.

Do you want to make that a separate issue?